plugins/woocommerce/src/Internal/Admin/ShippingLabelBanner.php

```diff
@@ -84,7 +73,7 @@ class_exists( '\Automattic\WCTax\Loader' );
 	 * @param \WP_Post $post Current post object.
 	 */
 	public function add_meta_boxes( $post_type, $post ) {
-		if ( 'shop_order' !== $post_type ) {
+		if ( ! OrderUtil::is_order_edit_screen() ) {
```

Comment:
Due to this change, then the DocBlock and parameter naming isn't always 100% correct. For HPOS orders, $post becomes `Automattic\WooCommerce\Admin\Overrides\Order` aka `WC_Order` and not `\WP_Post`.

I found this from a PHP warning about `Function ID was called incorrectly`. TL;DR: `$post->ID` further down should be called as `->get_id()` when it's not WP_Post 😄 

```
PHP Notice:  Function ID was called <strong>incorrectly</strong>. Order properties should not be accessed directly. Backtrace: do_action('load-woocommerce_page_wc-orders'), WP_Hook-&gt;do_action, WP_Hook-&gt;apply_filters, Automattic\WooCommerce\Internal\Admin\Orders\PageController-&gt;__call, call_user_func_array, Automattic\WooCommerce\Internal\Admin\Orders\PageController-&gt;handle_load_page_action, Automattic\WooCommerce\Internal\Admin\Orders\PageController-&gt;setup_action_edit_order, Automattic\WooCommerce\Internal\Admin\Orders\PageController-&gt;prepare_order_edit_form, Automattic\WooCommerce\Internal\Admin\Orders\Edit-&gt;setup, do_action('add_meta_boxes'), WP_Hook-&gt;do_action, WP_Hook-&gt;apply_filters, Automattic\WooCommerce\Internal\Admin\ShippingLabelBanner-&gt;add_meta_boxes, WC_Abstract_Legacy_Order-&gt;__get, wc_doing_it_wrong Please see <a href="https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/">Debugging in WordPress</a> for more information. (This message was added in versi in /var/www/html/wp-includes/functions.php on line 6085
```

---

**Context**: WC has "re-created" `do_action( 'add_meta_boxes' ... )` for HPOS since it's not a CPT anymore (`Automattic\WooCommerce\Internal\Admin\Orders\Edit::setup`)

Reply:
Addressed in b0e89e45f9932e2a7e6e9a174aefa0386a443e90

Reply:
Addressed in 3ed60a42d764eeca86234cd823b3c7bfc97077cc

Reply:
Awesome, thank you Sam!